packages/gatsby-plugin-page-creator/src/get-collection-route-params.ts

```diff
@@ -0,0 +1,24 @@
+import path from "path"
+
+// This extracts params from its filePath counerpart
+// and returns an object of it's matches.
+// e.g.,
+//   /foo/{id}, /foo/123 => {id: 123}
+export function getCollectionRouteParams(
+  filePath: string,
+  urlPath: string
+): Record<string, string> {
+  const params = {}
+  // remove the starting path to simplify the loop
+  const fileParts = filePath.split(path.sep)
+  const urlParts = urlPath.split(path.sep)
```

Comment:
Windows-related: I always get `filePath` and `urlPath` with regular slashes here (e.g. `/posts/{fields__slug}/` vs. `\posts\{fields__slug}\` as you would expect on Windows). So `path.sep` doesn't do what expected here. 

It happens because those paths are produced by `createPath` utility here:

https://github.com/gatsbyjs/gatsby/blob/9720e82acc0afc462bd93f25f657d49b37f2da10/packages/gatsby-plugin-page-creator/src/create-pages-from-collection-builder.ts#L61-L63

And this utility actually always joins path segments by `/` using `path.posix.join`:

https://github.com/gatsbyjs/gatsby/blob/9720e82acc0afc462bd93f25f657d49b37f2da10/packages/gatsby-page-utils/src/create-path.js#L4-L9

So this method always returns empty `params` for me but surprisingly it doesn't seem to affect anything? Pages are still created and routing seems to be working. So what do we use those params for?

Reply:
Ah so it makes sense that this wouldn't _fail_. What it does is it extracts the url segments and puts it to the component on props.params. So I'm guessing for windows users, props.params in a collection route